src/game.cpp

```diff
@@ -10459,6 +10459,23 @@ bool game::plmove( int dx, int dy, int dz )
         return true;
     }
 
+    //Wooden Fence Gate: vault over it if we are running, open it if we are walking
+    if( m.ter( dest_loc ) == t_fencegate_c ) {
```

Comment:
I don't think this should be specific to the `t_fencegate_c` terrain type.  Any terrain that is both openable and walkable should probably be handled similarly.
(I realise there are already other hardcoded terrain types in this function, but I would argue they should also be phased out)

Reply:
If walkable does not include regular doors and chain gates and such, that would make good sense yes.

Reply:
Can you point me in a direction for the terrain flags? I can't seem to find "openable" terrain/furniture by myself.

Reply:
The easy way to implement this (I think) would be to just mimic the logic to open doors from further down, but do it here also if the player is not running.  If nothing there is openable then `open_door` will return `false` and the function will proceed as before.  You'd lose the "vault over" message, but maybe that doesn't fit in this PR anyway.

Reply:
If you really want to test for something openable without actually opening it yet then you can call `open_door` with `check_only=true`.

Reply:
I can remove the messaging if you like, that wouldn't be a problem.

Reply:
The concern I have about the message is that it might turn out to be untrue if the move fails for some reason (again, no specific failure mode in mind).  Also it seems inappropriate for characters with the bad knees trait, or two broken arms.

Reply:
That's reasonable.